modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/ODBProxy.scala

```diff
@@ -4,16 +4,107 @@ import edu.gemini.pot.sp.SPObservationID
     SeqExecService.client(loc).sequence(oid).leftMap(SeqexecFailure.ODBSeqError)
 
+  val datasetStart: (SPObservationID, String, ImageFileId) => SeqAction[Boolean] = cmds.datasetStart
+  val datasetComplete: (SPObservationID, String, ImageFileId) => SeqAction[Boolean] = cmds.datasetComplete
+  val obsAbort: (SPObservationID, String) => SeqAction[Boolean] = cmds.obsAbort
+  val sequenceEnd: (SPObservationID) => SeqAction[Boolean] = cmds.sequenceEnd
+  val sequenceStart: (SPObservationID, ImageFileId) => SeqAction[Boolean] = cmds.sequenceStart
+  val obsContinue: (SPObservationID) => SeqAction[Boolean] = cmds.obsContinue
+  val obsPause: (SPObservationID, String) => SeqAction[Boolean] = cmds.obsPause
+  val obsStop: (SPObservationID, String) => SeqAction[Boolean] = cmds.obsStop
+
+}
+
+object ODBProxy {
+  trait OdbCommands {
+    def datasetStart(obsId: SPObservationID, dataId: String, fileId: ImageFileId): SeqAction[Boolean]
+    def datasetComplete(obsId: SPObservationID, dataId: String, fileId: ImageFileId): SeqAction[Boolean]
+    def obsAbort(obsId: SPObservationID, reason: String): SeqAction[Boolean]
+    def sequenceEnd(obsId: SPObservationID): SeqAction[Boolean]
+    def sequenceStart(obsId: SPObservationID, fileId: ImageFileId): SeqAction[Boolean]
+    def obsContinue(obsId: SPObservationID): SeqAction[Boolean]
+    def obsPause(obsId: SPObservationID, reason: String): SeqAction[Boolean]
+    def obsStop(obsId: SPObservationID, reason: String): SeqAction[Boolean]
+  }
+
+  object DummyOdbCommands extends OdbCommands {
+    override def datasetStart(obsId: SPObservationID, dataId: String, fileId: ImageFileId): SeqAction[Boolean] = SeqAction(false)
+    override def datasetComplete(obsId: SPObservationID, dataId: String, fileId: ImageFileId): SeqAction[Boolean] = SeqAction(false)
+    override def obsAbort(obsId: SPObservationID, reason: String): SeqAction[Boolean] = SeqAction(false)
+    override def sequenceEnd(obsId: SPObservationID): SeqAction[Boolean] = SeqAction(false)
+    override def sequenceStart(obsId: SPObservationID, fileId: ImageFileId): SeqAction[Boolean] = SeqAction(false)
+    override def obsContinue(obsId: SPObservationID): SeqAction[Boolean] = SeqAction(false)
+    override def obsPause(obsId: SPObservationID, reason: String): SeqAction[Boolean] = SeqAction(false)
+    override def obsStop(obsId: SPObservationID, reason: String): SeqAction[Boolean] = SeqAction(false)
+  }
+
+  case class OdbCommandsImpl(host: Peer) extends OdbCommands {
+    val xmlrpcClient = new WDBA_XmlRpc_SessionClient(host.host, host.port.toString)
+    val sessionName = "sessionQueue"
+
+    override def datasetStart(obsId: SPObservationID, dataId: String, fileId: ImageFileId): SeqAction[Boolean] = EitherT(
+      Task(
+        xmlrpcClient.datasetStart(sessionName, obsId.stringValue, dataId, fileId.toString)
+      ).attempt.map(_.leftMap(e => SeqexecFailure.SeqexecException(e)))
```

Comment:
Attempt does not run the sequence. It puts possible exception in the output, as a disjunction